Potestas/Potestas/Observations/Comparers/EqualityComparers/ObservationPointEqualityComparer.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+
+namespace Potestas.Observations.Comparers.EqualityComparers
+{
+    public class ObservationPointEqualityComparer<T> : BaseEqualityComparer<T> where T : IEnergyObservation
+    {
+        public override bool Equals(T xObservation, T yObservation)
+        {
+            var baseEqualityCompareResult = BaseEqualityCompare(xObservation, yObservation);
+
+            if (baseEqualityCompareResult.HasValue)
+            {
+                return baseEqualityCompareResult.Value;
+            }
+
+            if (double.IsNaN(xObservation.ObservationPoint.X) && double.IsNaN(xObservation.ObservationPoint.Y))
+            {
+                return double.IsNaN(yObservation.ObservationPoint.X) && double.IsNaN(yObservation.ObservationPoint.Y);
+            }
+
+            if (double.IsNaN(xObservation.ObservationPoint.X) && !double.IsNaN(xObservation.ObservationPoint.Y))
+            {
+                return double.IsNaN(yObservation.ObservationPoint.X) && !double.IsNaN(yObservation.ObservationPoint.Y);
+            }
+
+            if (!double.IsNaN(xObservation.ObservationPoint.X) && double.IsNaN(xObservation.ObservationPoint.Y))
+            {
+                return !double.IsNaN(yObservation.ObservationPoint.X) && double.IsNaN(yObservation.ObservationPoint.Y);
+            }
```

Comment:
Can we just create a helper  method or class, kind of PrecisionComparer, that would compare two doubles considering the precision? Then here we would just call this method for X and Y.

Reply:
done https://github.com/ChristinaGolovach/.NET.Mentoring.L1/pull/1/commits/1dcade325f0ba826f513eb47c7b0e93d0e6a204b